src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarReader.cs

```diff
@@ -198,6 +270,60 @@ private void Dispose(bool disposing)
             }
         }
 
+        // Asynchronously disposes the current instance.
+        // If 'disposing' is 'false', the method was called from the finalizer.
+        private async ValueTask DisposeAsync(bool disposing)
```

Comment:
Since the type is `sealed` it makes sense to do all of the DisposeAsync logic in DisposeAsync, instead of `DisposeAsyncCore()` (which this method should have been called for the basic async dispose pattern).

Similarly, `Dispose(bool)` can be merged into `Dispose()`, and all of the `GC.SuppressFinalize(this)`s can be removed.